src/fu-engine-helper.c

```diff
@@ -133,3 +135,53 @@ fu_engine_update_motd(FuEngine *self, GError **error)
 	g_debug("writing motd target %s", target);
 	return g_file_set_contents(target, str->str, str->len, error);
 }
+
+gboolean
+fu_engine_update_devices_file(FuEngine *self, GError **error)
+{
+	gsize len;
+	g_autoptr(JsonBuilder) builder = NULL;
+	g_autoptr(JsonGenerator) generator = NULL;
+	g_autoptr(JsonNode) root = NULL;
+	g_autoptr(GPtrArray) devices = NULL;
+	g_autofree gchar *data = NULL;
+	g_autofree gchar *directory = NULL;
+	g_autofree gchar *target = NULL;
+
+	builder = json_builder_new();
+	json_builder_begin_object(builder);
+	json_builder_set_member_name(builder, "Devices");
+	json_builder_begin_array(builder);
+	devices = fu_engine_get_devices(self, NULL);
+	if (devices != NULL) {
+		json_builder_begin_array(builder);
+		for (guint i = 0; i < devices->len; i++) {
+			FwupdDevice *dev = g_ptr_array_index(devices, i);
+			json_builder_begin_object(builder);
+			fwupd_device_to_json(dev, builder);
+			json_builder_end_object(builder);
+		}
+		json_builder_end_array(builder);
+	}
+	json_builder_end_array(builder);
+	json_builder_end_object(builder);
+
+	root = json_builder_get_root(builder);
+	generator = json_generator_new();
+	json_generator_set_pretty(generator, TRUE);
+	json_generator_set_root(generator, root);
+	data = json_generator_to_data(generator, &len);
+	if (data == NULL) {
+		g_set_error_literal(error,
+				    FWUPD_ERROR,
+				    FWUPD_ERROR_INTERNAL,
+				    "Failed to convert to JSON string");
+		return FALSE;
+	}
+
+	directory = fu_common_get_path(FU_PATH_KIND_CACHEDIR_PKG);
```

Comment:
What about storing in the `/run` directory instead of the `/var/cache`?

Reply:
@superm1 ? this feels like the wong things to do from a FHS but it is appealing because it's cleared on reboot.

Reply:
I think /var/cache seems most appropriate tbh.

Reply:
Ok